misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/PartialCoverageModifiedCounter.java

```diff
@@ -0,0 +1,82 @@
+package org.ballerinalang.test.runtime.entity;
+
+import org.jacoco.core.analysis.ICounter;
+
+/**
+ * Represents a counter that eliminates partially covered counter status and coverts it to fully covered status.
+ *
+ */
+public class PartialCoverageModifiedCounter implements ICounter {
+
+    private int covered;
+    private int missed;
+
+    public PartialCoverageModifiedCounter(ICounter prevCounter) {
+        this.covered = prevCounter.getCoveredCount();
+        this.missed = prevCounter.getMissedCount();
+        modifyCoverageNumbers();
+    }
+
+    /**
+     * Modify the covered and missed numbers in cases the counter status is calculated as PARTLY_COVERED.
+     * It converts the counter status to FULLY_COVERED.
+     */
+    public void modifyCoverageNumbers() {
```

Comment:
Any reason to make this `public`. Rule of thumb is to make it more restricted first(such as private) then consider to make less restrict base on the usage and future changes.

Reply:
Will convert it to private

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/28767/commits/b33977232ac9c95a1b70e8307c3c9eb5133bde89